### ISSUE TITLE
TidalPy.RadialSolver.solver missing after crash

### ISSUE THREAD

**Issue by nlwagner** [jrenaud90/TidalPy#50]
I'll try to detail the conditions that led up to this crash as best I can.

I was testing radial_solver with much finer layers and more property variations than previously. I'm currently using Python 3.9 and Spyder version 5.5.1 through conda. 

After rebooting Spyder, however, I'm met with this error:
```
runfile('/Users/nlwagner25/Desktop/PYTHON/TidalPy/testing_AK_model_elastic.py', wdir='/Users/nlwagner25/Desktop/PYTHON/TidalPy')
2024-02-16 14:33:05(+00:00:28::286448) - INFO     : TidalPy initializing...
2024-02-16 14:33:05(+00:00:28::287662) - INFO     : TidalPy initialization complete.
Traceback (most recent call last):

  File ~/opt/anaconda3/envs/tidalpy/lib/python3.9/site-packages/spyder_kernels/py3compat.py:356 in compat_exec
    exec(code, globals, locals)

  File ~/Desktop/PYTHON/TidalPy/testing_AK_model_elastic.py:15
    from TidalPy.RadialSolver import radial_solver

  File ~/Desktop/PYTHON/TidalPy/TidalPy/RadialSolver/__init__.py:1
    from TidalPy.RadialSolver.solver import radial_solver

ModuleNotFoundError: No module named 'TidalPy.RadialSolver.solver'
```

I tried reinstalling TidalPy through Pip, but no luck. I'm completely deleting and recreating my conda environment and will update with a comment once that finishes.

**Comment by nlwagner** [comment]
After a clean conda environment install I'm receiving the same error.

**Comment by nlwagner** [comment]
After some testing, it seems this error is only within Spyder. Jupyter notebooks still work fine. Tested with the included demo notebooks within /Demos.

**Comment by jrenaud90** [comment]
I will play around with spyder to see if I can run into the issue. I think my first question/concern is what led to that initial crash

**Comment by nlwagner** [comment]
So running the script from the command line results in the same error:
```nlwagner25@Nicks-Macbook : ~/Desktop/PYTHON/TidalPy                                                                                                                                             
$ python testing_AK_model_elastic.py                                                                                                                                                            
2024-02-22 11:52:32(+00:00:00::825509) - INFO     : TidalPy initializing...                                                                                                                     
2024-02-22 11:52:32(+00:00:00::825607) - INFO     : TidalPy initialization complete.                                                                                                            
Traceback (most recent call last):                                                                                                                                                              
  File "/Users/nlwagner25/Desktop/PYTHON/TidalPy/testing_AK_model_elastic.py", line 15, in <module>                                                                                             
    from TidalPy.RadialSolver import radial_solver                                                                                                                                              
  File "/Users/nlwagner25/Desktop/PYTHON/TidalPy/TidalPy/RadialSolver/__init__.py", line 1, in <module>                                                                                         
    from TidalPy.RadialSolver.solver import radial_solver                                                                                                                                       
ModuleNotFoundError: No module named 'TidalPy.RadialSolver.solver'                                                                                                                              
(TidalPy)
```

The best I can remember is I was just testing radial_solver for a model I had. The file is testing_AK_model_elastic.py in my forked repository if you can access it. It hasn't been modified since the crash.

In testing a new conda env I run the following:
```
 conda create -n tidalpy2 python="3.11" numpy spyder jupyter matplotlib scipy

 conda activate tidalpy2

 Pip install TidalPy

 python testing_AK_model_elastic.py 

```
The last line running the script results in the error posted before.

**Comment by jrenaud90** [comment]
Hey @nlwagner! 

Okay I think I have found a few issues.

- The "radial_solver" not found is likely because you are running your script from the same directory that TidalPy is located. Python is trying to import the local directory's tidalpy instead of the one that was pre-installed. This will cause a host of problems.
    - Fix: Move your script (and any required aux files) to a different directory. 
- After doing this I do get the script to run but get the crash. One issue I see is that you are passing way more `is_static_by_layer` "layer_types" etc than you actually have layers. The `is_static_by_layer`, `is_incompressible_by_layer`, `upper_radius_by_layer` should all be the same size. You only need layers for liquid-solid interfaces and large density jumps. So if you have 100 radius "slices" (`radius_array.size == 100`) but only the lower half are liquid then you only have 2 "layers". So `layer_types=('liquid', 'solid')`.
    - I think this is causing the crash. TidalPy is trying to pull a value of `upper_radius_by_layer[>6]` which does not exist.
    - I already have a fix coming to v0.5.2 for this (well not a fix, but a proper error will be raised rather than a crash).
- I think you don't want to perform the reverse step you are doing early on. I am looking at the values of "mu" and the liquid layer is near the surface not the core. TidalPy assumes `radius_array[0]` is near the planet center rather than on the surface. 
- Complex shear is not being calculated correctly. If you look at the values they are all un-initialized memory. I noticed that you are setting viscosity = inf. I was wondering if this was the problem but since you are using an elastic model it should not matter. I am a little stumped what is happening here but I can get it to work if I throw `np.copy`s around:
    - `elastic_rheology.vectorize_modulus_viscosity(forcing_frequency, np.copy(shear_array), np.copy(viscosity_array), complex_shear)`
    - This seems to be a workaround for now but definitely do some print()s on complex_shear before you pass it to radial_solver to make sure it makes sense. I will work on what is going on there.
    
Cheers,

**Comment by nlwagner** [comment]
Thanks so much Joe. Sorry for the late reply, been focusing on LPSC related things. I followed your advice and I now have it working! I'm glad it was a simple issue. I definitely misunderstood the layer types input, I thought that had to be the same size as my input array.

I'll let you know if I run into any other errors I can't overcome :)
